sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,60 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
```

Comment:
use `enum class`? This will hide HttpMethod's members visibility from the global scope, and they will only be accessible as `http_method::HttpMethod::PATCH`, and not as `http_method::PATCH`, so then you don't need a namespace to achieve this effect.

Reply:
Any downsides of using `enum class` vs. just `enum`?

Reply:
`enum` is just there because initially C++ evolved from C, and in C historically all the enum members you define in enum are available without the enum prefix. This is why people explicitly prefix enum values as `enum FileOpenMode { FILEPOENMODE_READ, FILEOPENMODE_WRITE }`. C++ has to maintain backwards compatibilty with C89. And C89 at a time had it for the reason no one envisioned it would go all this way at a time, and compilers were simpler back then. This is why enum name rules work the same way in C++, if you use just `enum`. And in C++11 they added the `enum class` - with better naming rules, and it does not break compatibility with C89. So now you can declare `enum class FileOpenMode { READ, WRITE }`. But should you include legacy C header from C++ code, which has just plain enums and not enum classes, and C++ can compile that code as well.